addons/sale/models/sale.py

```diff
@@ -264,6 +264,17 @@ def create(self, vals):
         return result
 
     @api.multi
+    def copy_data(self, default=None):
+        if default is None:
+            default = {}
+        if 'order_line' not in default:
+            default['order_line'] = []
+            for line in self.order_line:
```

Comment:
I bet you can do a one-liner with this code to simplify it a bit :) . Use filtered and /or list comprehension.